dask_ml/model_selection/_incremental.py

```diff
@@ -459,43 +463,61 @@ def _get_params(self):
         """
         return ParameterGrid(self.parameters)
 
-    def _get_history_results(self, results):
-        # type: (Results) -> Dict
-        """Construct the CV results.
-
-        Has the following keys:
-
-        * params
-        * test_score
-        * mean_test_score
-        * rank_test_score
-        * mean_partial_fit_time
-        * std_partial_fit_time
-        * mean_score_time
-        * std_score_time
-        * partial_fit_calls
-        * model_id
-        """
-        info, model, history = results
-        key = operator.itemgetter("model_id")
-        hist2 = sorted(history, key=key)
-        return hist2
-
-    def _get_best(self, results, history_results):
-        # type: (Dict, Dict) -> Estimator
-        """Select the best estimator from the set of estimators."""
-        best_model_id = first(results.info)
-        key = operator.itemgetter("model_id")
-        best_index = -1
-        # history_results is sorted by (model_id, partial_fit_calls)
-        # best is the model_id with the highest partial fit calls
-        for k, v in itertools.groupby(history_results, key=key):
-            v = list(v)
-            best_index += len(v)
-            if k == best_model_id:
-                break
-
-        return results.models[best_model_id], best_index
+    def _get_cv_results(self, history, model_hist):
+        cv_results = {}
+        best_scores = {}
+        best_scores = {k: hist[-1]["score"] for k, hist in model_hist.items()}
+
+        cv_results = {}
+        for k, hist in model_hist.items():
+            pf_times = list(toolz.pluck("partial_fit_time", hist))
+            score_times = list(toolz.pluck("score_time", hist))
+            cv_results[k] = {
+                "mean_partial_fit_time": np.mean(pf_times),
+                "mean_score_time": np.mean(score_times),
+                "std_partial_fit_time": np.std(pf_times),
+                "std_score_time": np.std(score_times),
+                "test_score": best_scores[k],
+                "model_id": k,
+                "params": hist[0]["params"],
+                "partial_fit_calls": hist[-1]["partial_fit_calls"],
+            }
+        cv_results = list(cv_results.values())  # list of dicts
+        cv_results = {k: [res[k] for res in cv_results] for k in cv_results[0]}
+
+        # Every model will have the same params because this class uses either
+        # ParameterSampler or ParameterGrid
+        cv_results.update(
+            {
+                "param_" + k: v
+                for params in cv_results["params"]
+                for k, v in params.items()
+            }
+        )
+        cv_results = {k: np.array(v) for k, v in cv_results.items()}
+
+        order = (-1 * cv_results["test_score"]).argsort()
```

Comment:
Could replace this with

```python
        cv_results = {k: np.array(v) for k, v in cv_results.items()}
        cv_results["rank_test_score"] = scipy.stats.rankdata(
            -cv_results["test_score"], method="min"
        ).astype(int)
        return cv_results
```

With `method="min"`, we assign ties the same rank. I'm not sure what scikit-learn does in the case of ties, but this seems more correct. It makes the assertion below that `assert search.cv_results_["rank_test_score"][search.best_index_] == 1` always true.

Reply:
Scikit-learn also uses rankdata https://github.com/scikit-learn/scikit-learn/blob/e0e738760625129be71839fac2dad9325fb90225/sklearn/model_selection/_search.py#L24